pkg/broadcaster/broadcaster.go

```diff
@@ -0,0 +1,77 @@
+package broadcaster
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+// Broadcaster is a simplified Broker allowing clients to subscribe while
+// itself behaving as a io.Writer.
+type Broadcaster struct {
+	mu sync.Mutex
+	// NOTE: As we use a map, the broadcast order is randomized.
+	// Consider using a slice instead to be deterministic.
+	clients map[io.WriteCloser]struct{}
+}
+
+func NewBroadcaster() *Broadcaster {
+	return &Broadcaster{
+		clients: map[io.WriteCloser]struct{}{},
+	}
+}
+
+// Surface the lock as Pause/Unpause to freeze the broadcast while we make a copy of historical data.
+func (b *Broadcaster) Pause()   { b.mu.Lock() }
+func (b *Broadcaster) UnPause() { b.mu.Unlock() }
+
+// If broadcaster is closed, no nothing.
+func (b *Broadcaster) Subscribe(w io.WriteCloser) {
```

Comment:
Indeed, but then it wouldn't be an agnostic broadcaster anymore, it would be much more specific to the use-case.
Exposing Pause/Unpause is not ideal, but it makes the code way simpler and still allows for the package to be generic enough it could be re-used somewhere else.